src/Akeneo/Git/PullRequestCreator.php

```diff
@@ -31,64 +38,72 @@ class PullRequestCreator
+        EventDispatcherInterface $eventDispatcher,
+        $fork_owner,
+        $owner,
+        $repository
+    ) {
+        $this->executor        = $executor;
+        $this->client          = $client;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->fork_owner      = $fork_owner;
+        $this->owner           = $owner;
+        $this->repository      = $repository;
     }
 
     /**
+     * Create a new Pull Request
+     *
      * @param string $baseBranch
      * @param string $baseDir
      * @param string $projectDir
      */
     public function create($baseBranch, $baseDir, $projectDir)
     {
         $branch = $baseBranch.'-'.(new \DateTime())->format('Y-m-d-H-i');
-        $cmd = sprintf(
-            'cd %s && ' .
-            'git checkout -b crowdin/%s && ' .
-            'git add .',
-            $projectDir,
-            $branch
-        );
-        $this->executor->execute($cmd);
-
-        $cmd = sprintf(
-            'cd %s git && git commit -m "[Crowdin] Updated translations" && git push origin crowdin/%s',
-            $projectDir,
-            $branch
-        );
-        $this->executor->execute($cmd);
-
-        $this->client->api('pr')->create(
-            $this->owner,
-            $this->repository,
-            [
-                'head'  => sprintf('%s:crowdin/%s', $this->fork_owner, $branch),
-                'base'  => $baseBranch,
-                'title' => 'Update translations from Crowdin',
-                'body'  => 'Updated on ' . $branch,
-            ]
-        );
-
-        $cmd = sprintf('cd %s/ && rm -rf *.zip', $baseDir);
-        $this->executor->execute($cmd);
-
-        $cmd = sprintf(
-            'cd %s && ' .
-            'git checkout master',
-            $projectDir
-        );
-        $this->executor->execute($cmd);
+
+        $this->eventDispatcher->dispatch(Events::PRE_GITHUB_CREATE_PR, new GenericEvent(null, [
+            'name' => $branch,
+            'branch' => $baseBranch
+        ]));
+
+        $this->executor->execute(sprintf('cd %s && git checkout -B crowdin/%s', $projectDir, $branch));
+
+        $this->executor->execute(sprintf('cd %s && git add .', $projectDir));
+
+        $this->executor->execute(sprintf('cd %s && git commit -m "[Crowdin] Updated translations"', $projectDir));
+
+        $this->executor->execute(sprintf('cd %s && git push origin crowdin/%s', $projectDir, $branch));
+
+        try {
+            $this->client->api('pr')->create(
+                $this->owner,
+                $this->repository,
+                [
+                    'head' => sprintf('%s:crowdin/%s', $this->fork_owner, $branch),
+                    'base' => $baseBranch,
+                    'title' => 'Update translations from Crowdin',
+                    'body' => 'Updated on ' . $branch,
+                ]
+            );
+        } catch (ValidationFailedException $exception) {
+            // This append when there is no modifications.
```

Comment:
Means the PR will not be created?

I would raise a dedicated business event here to say that there is nothing to update and based on this event I would display an message "all translations are already up to date for the branch 1.x"